### PR TITLE
Cleanup Lint Errors and Warnings

### PR DESCRIPTION
**Pull request by outragedhuman**
Cleans up several lint errors and warnings including unused vars, unused imports, and "any" types.

### CODE REVIEW

File: cypress/suites/login.suite.ts [hunk 1]
```diff
@@ -239,6 +235,30 @@ export default {
         expectToBeOnDashboard();
       });
 
+      /* eslint-disable-next-line mocha-cleanup/asserts-limit */
+      it.only("should allow a user to go back to the beginning from any step", () => {
```
Comment (outragedhuman) [inline cypress/suites/login.suite.ts:239]: remove only

File: cypress/suites/login.suite.ts [hunk 2]
```diff
@@ -239,6 +235,30 @@ export default {
...
+        cy.get("#onboarding_back_button").trigger("click");
+
+        // go to third step
+        expectClickGenerateWalletButton();
+        cy.get("textarea").invoke("val").then(stringToWordArray).as("seedWords");
+        cy.get("label").contains("I have backed up my seed phrase").click();
+        cy.get("#onboarding_back_button").trigger("click");
+
+        // go to final step
+        cy.get("textarea").invoke("val").then(stringToWordArray).as("seedWords");
```
Comment (outragedhuman) [inline cypress/suites/login.suite.ts:251]: would be pretty slick to make a function which could take the user to a step based on a passed in argument so these could look like:

`goToOnboardingStep(2)` for the second onboarding step

File: src/contexts/MyTxContext/Provider.tsx [hunk 3]
```diff
@@ -16,7 +17,9 @@ const MyTxProvider: React.FC = ({ children }) => {
     }
 
     const tx = await getMyTxs(accountRs);
```
Comment (outragedhuman) [inline src/contexts/MyTxContext/Provider.tsx:19]: try catch?

File: src/utils/api/constants.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 // TODO: Make this configurable based on prod/dev run status
-export const BASEURL = "http://localhost:3080/nxt?"; // proxy runs on 3080 by default
+export const BASEURL = "https://nodes.jup.io/nxt?"; // prox URL
```
Comment (outragedhuman) [inline src/utils/api/constants.ts:2]: revert to localhost, this was done for a build test

File: src/views/Dashboard/components/Widgets/SendWidget/SendWidget.tsx [hunk 5]
```diff
@@ -88,15 +88,15 @@ const StyledWidgetHeading = styled(Typography)(() => ({
 
 const StyledToAddressInput = styled(JUPInput, {
   shouldForwardProp: (prop) => prop !== "onChange",
-})<{ onChange?: (e: any) => void }>(({ onChange }) => ({
+})<{ onChange?: (e: React.SyntheticEvent) => void }>(() => ({
```
Comment (outragedhuman) [inline src/views/Dashboard/components/Widgets/SendWidget/SendWidget.tsx:91]: create an issue for specifically discussing this one with V, I wonder if passing `{...other}` through to the component would fix this in the styling

**Review by outragedhuman — COMMENTED** [review]
self review comments

File: cypress/suites/login.suite.ts (hunk 1 above)
Comment (outragedhuman) [reply cypress/suites/login.suite.ts:239]: fixed

File: src/contexts/MyTxContext/Provider.tsx (hunk 3 above)
Comment (outragedhuman) [reply src/contexts/MyTxContext/Provider.tsx:19]: fixed

File: src/utils/api/constants.ts (hunk 4 above)
Comment (outragedhuman) [reply src/utils/api/constants.ts:2]: fixed

File: src/views/Dashboard/components/Widgets/SendWidget/SendWidget.tsx (hunk 5 above)
Comment (outragedhuman) [reply src/views/Dashboard/components/Widgets/SendWidget/SendWidget.tsx:91]: https://github.com/Up-North-Coding/jovian/issues/100

File: proxy/server.js [hunk 6]
```diff
@@ -13,8 +13,8 @@ const app = express();
 process.env.UV_THREADPOOL_SIZE = 20;
 
 let counter = 0;
-// const BASEURL = "https://nodes.jup.io"; // prod url
-const BASEURL = "https://test.jup.io"; // test url
+const BASEURL = "https://nodes.jup.io"; // prod url
```
Comment (vwalsh) [inline proxy/server.js:16]: i thought the proxy was to stay on the test endpoint? would at least like to discuss this

File: src/components/JUPAssetSearchBox/JUPAssetSearchBox.tsx [hunk 7]
```diff
@@ -70,8 +72,7 @@ const JUPAssetSearchBox: React.FC<IJUPAssetSearchBoxProps> = ({ fetchFn }) => {
     <Autocomplete
       freeSolo
       options={searchBoxResults.map((searchBoxValue) => searchBoxValue)}
-      // MUST: fix magic number? currently splits the value and takes the asset id portion (first index), might be able to do this more cleanly
-      onChange={(e, value: any) => value && fetchFn(value.split("-")[1].trim())}
+      onChange={(e, value) => value && fetchFn(value.split("-")[AssetIdIndex].trim())}
```
Comment (vwalsh) [inline src/components/JUPAssetSearchBox/JUPAssetSearchBox.tsx:75]: fyi, index 1 is not the first index it's the 2nd. 0 is the first index in an array.

File: src/contexts/APIRouterContext/Provider.tsx [hunk 8]
```diff
@@ -244,13 +244,13 @@ const APIRouterProvider: React.FC = ({ children }) => {
...
 
   const handleSubmitSecret = useCallback(async () => {
-    let result;
     try {
       if (afterSecretCB.current === undefined) {
         throw new Error(`handleSubmitSecret afterSecretCB.current is undefined: ${afterSecretCB.current}`);
       }
 
-      result = await afterSecretCB.current(userSecretInput);
+      // TODO: should consider doing something with the return of this function call
```
Comment (vwalsh) [inline src/contexts/APIRouterContext/Provider.tsx:252]: i think the try/catch nature here is what we care about. if there is a problem, throw an error.

a return could also be helpful, unsure yet what it needs to be used for.

File: src/utils/api/constants.ts [hunk 9]
```diff
@@ -1,4 +1,5 @@
 // TODO: Make this configurable based on prod/dev run status
+// export const BASEURL = "https://nodes.jup.io/nxt?"; // prod URL
 export const BASEURL = "http://localhost:3080/nxt?"; // proxy runs on 3080 by default
```
Comment (vwalsh) [inline src/utils/api/constants.ts:3]: we can define a PROD vs DEV variable that the vitejs server will replace here. probably a safer way than assuming we had it set properly for a delivery.

File: src/utils/api/getBlockchainTransactions.ts [hunk 10]
```diff
@@ -30,7 +30,7 @@ async function getBlockchainTransactions(account: string): Promise<false | IGetB
     console.error("error getBlockchainTransactions():", e);
     return false;
   }
-  return result.transactions;
+  return result;
```
Comment (vwalsh) [inline src/utils/api/getBlockchainTransactions.ts:33]: would like to know why this change exists. seems sus to me.

File: src/utils/common/txBuilder.ts [hunk 11]
```diff
@@ -11,7 +11,7 @@ import {
   standardTransactionVersion,
 } from "./constants";
 
-export function buildTx({ ...args }: any) {
+export function buildTx({ ...args }) {
```
Comment (vwalsh) [inline src/utils/common/txBuilder.ts:14]: this could still have a type consideration because i believe it's implying `any` and that implication could be bad ;)

**Review by vwalsh — COMMENTED** [review]
see comments

File: src/components/JUPAssetSearchBox/JUPAssetSearchBox.tsx (hunk 7 above)
Comment (outragedhuman) [reply src/components/JUPAssetSearchBox/JUPAssetSearchBox.tsx:75]: all good, working as intended

File: src/contexts/APIRouterContext/Provider.tsx (hunk 8 above)
Comment (outragedhuman) [reply src/contexts/APIRouterContext/Provider.tsx:252]: yea same here, just seems like it needs something I just don't know what yet

File: src/utils/api/constants.ts (hunk 9 above)
Comment (outragedhuman) [reply src/utils/api/constants.ts:3]: not a high priority for me atm but I wouldn't stop you from doing it

File: src/utils/api/getBlockchainTransactions.ts (hunk 10 above)
Comment (outragedhuman) [reply src/utils/api/getBlockchainTransactions.ts:33]: was part of eslint prod cleanup, when I fixed an `any` i had to make some minor adjustments to types which meant changing what I passed down

File: src/utils/common/txBuilder.ts (hunk 11 above)
Comment (outragedhuman) [reply src/utils/common/txBuilder.ts:14]: i'll happily kick this can down the road until there's more tx types to be built

File: proxy/server.js (hunk 6 above)
Comment (outragedhuman) [reply proxy/server.js:16]: i had been trying to keep it as the prod URL each time I merged to master but that's probably no longer a good idea now that tests have been re-tailored to the testnet URL, so I'll keep it as the test URL moving forward

File: src/contexts/APIRouterContext/Provider.tsx (hunk 8 above)
Comment (vwalsh) [reply src/contexts/APIRouterContext/Provider.tsx:252]: check if typescript will protect us by changing the functions for the callbacks to return a `Promise<undefined>` only. should help
